saleor/graphql/product/mutations/products.py

```diff
@@ -1158,6 +1170,17 @@ def clean_input(
     ):
         cleaned_input = super().clean_input(info, instance, data)
 
+        weight = cleaned_input.get("weight")
+        if weight and weight.value < 0:
+            raise ValidationError(
+                {
+                    "weight": ValidationError(
+                        "Product variant can't have negative weight.",
+                        code=ProductErrorCode.INVALID.value,
+                    )
+                }
+            )
+
```

Comment:
can be extracted to something like `validate_weight` and used here and there as there are already 3 places with the same code

Reply:
But errors are different. Second, think is that we have more duplicated logic in `clean_input` in ProductVariantCreate, ProductTypeCreate, and ProcutCreate, We should consider refactor it globally instead of having some logic duplicated and some logic hidden in functions.